### PR TITLE
VTOL transition command

### PR DESCRIPTION
**Pull request by AndreasAntener**
For the VTOL auto flight demo we used a hacked mavlink command. This adds a properly setup command for VTOL transitions. Goals
- Listening to a new transition command (MAVLink PR coming)
- Pulling transition switch one level up

Waiting on VTOL cleanup and transition state flag. @tumbili, @LorenzMeier : is that going into the right direction?

TODO:
- [x] pull switch up for all types
- [x] implement vtol state message
- [x] reset transition command when dropping out of offboard
- [x] review
- [x] bench tested

### CODE REVIEW

**Comment by LorenzMeier** [comment]
I merged the MAVLink change. So we can update the MAVLink submodule and then this should build, right?

**Comment by AndreasAntener** [comment]
Yes. I still have to update the message implementation now that vtol cleanup is merged. After that it would be good to go.

Not sure if the message should already be added as a stream to one of the default sets. For us it would probably make sense to have it in the offboard set. What do you think?

**Comment by LorenzMeier** [comment]
Would add it everywhere, but at a very low rate for telemetry.

**Comment by AndreasAntener** [comment]
Ok, I'll finish the impl and bench test it next week.

File: src/modules/vtol_att_control/vtol_att_control_main.h [hunk 1]
```diff
@@ -162,6 +166,8 @@ class VtolAttitudeControl
 	struct vehicle_local_position_s		_local_pos;
 	struct airspeed_s 					_airspeed;			// airspeed
 	struct battery_status_s 			_batt_status; 		// battery status
+	struct vehicle_command_s			_vehicle_cmd;
+	struct vehicle_command_s			_vehicle_transition_cmd; // stores transition commands only
```
Comment (bansiesta) [inline src/modules/vtol_att_control/vtol_att_control_main.h]: I don't understand why there are two vehicle_command_s, it seems all the same, or am I not getting it?

**Comment by bansiesta** [comment]
I had one question, other than that it looks good.

File: src/modules/vtol_att_control/vtol_att_control_main.h (hunk 1 above)
Comment (AndreasAntener) [inline src/modules/vtol_att_control/vtol_att_control_main.h]: Any last command gets stored in `_vehicle_cmd`, only transition commands get stored in `_vehicle_transition_command` (and that's the only member which is actually used). I think I'll remove the later and add a distinct member variable for the transition command request.

The `_vehicle_cmd` is actually never used, I could make it local in the polling function. Do you usually avoid this to not increase the stack size of those functions?

**Comment by AndreasAntener** [comment]
I checked that the manual switch still works with a tiltrotor, tailsitter and standard config. Bench tested the mavlink command and internal states being reset if manual and offboard switches are used in combination.
If anybody else want's to have a look over the code that would be great, else I think it's good to go.

File: src/modules/mavlink/mavlink_messages.cpp [hunk 2]
```diff
@@ -2340,6 +2341,79 @@ class MavlinkStreamDistanceSensor : public MavlinkStream
...
+
+	void send(const hrt_abstime t)
+	{
+		struct vehicle_status_s status;
+
+		if (_status_sub->update(&status)) {
+			mavlink_vtol_state_t msg;
+
+			if (status.is_vtol)
+			{
```
Comment (SimonWilks) [inline src/modules/mavlink/mavlink_messages.cpp]: The braces should be on the same line as the if/else statements here and below.

File: src/modules/vtol_att_control/vtol_att_control_main.cpp [hunk 3]
```diff
@@ -483,10 +525,22 @@ void VtolAttitudeControl::task_main()
...
 		vehicle_airspeed_poll();
 		vehicle_battery_poll();
+		vehicle_cmd_poll();
 
 		// update the vtol state machine which decides which mode we are in
 		_vtol_type->update_vtol_state();
 
+		// reset transition command if not in offboard control
+		if (!_v_control_mode.flag_control_offboard_enabled)
+		{
```
Comment (SimonWilks) [inline src/modules/vtol_att_control/vtol_att_control_main.cpp]: No newline before the brace.

**Comment by SimonWilks** [comment]
Just some minor code formatting comments. I'll merge this this morning after they are addressed if there are no other comments.

**Comment by AndreasAntener** [comment]
Fixed code style, thanks for the review!

**Comment by SimonWilks** [comment]
Merged.

**Pull request closed without merging**